python/dgl/sampling/randomwalks.py

```diff
@@ -169,8 +169,17 @@ def random_walk(g, nodes, *, metapath=None, length=None, prob=None, restart_prob
         metapath = [g.get_etype_id(etype) for etype in metapath]
 
     gidx = g._graph
-    nodes = F.to_dgl_nd(utils.prepare_tensor(g, nodes, 'nodes'))
-    metapath = F.to_dgl_nd(utils.prepare_tensor(g, metapath, 'metapath'))
+    nodes = utils.prepare_tensor(g, nodes, 'nodes')
+    assert F.max(nodes, dim=0) < g.num_nodes(g.canonical_etypes[metapath[0]][0]), \
+        "Seed node ID exceeds the maximum number of nodes."
```

Comment:
This fixes that the GPU random walk code doesn't check it, while the CPU version does. I do the check on the python side because when it comes to the c++ code, `seeds` and `metapath` are in the GPU memory and we have to copy a lot of things to complete the check.

Reply:
Can we assert that in the GPU kernel? This is related to the discussion in #3853 .

Reply:
To complete this check, we have to get the maximum number of the start node. For example, in the CPU version:
```c++
const int64_t begin_ntype = hg->meta_graph()->FindEdge(metapath_data[0]).first;
const int64_t max_nodes = hg->NumVertices(begin_ntype);
```
For GPU sampling, the `metapath_data` is on the GPU and we have to copy it back to CPU to get `max_nodes`.
Here on the Python side, we can do the check before moving `metapath` to GPU to save the copy.